src/test/java/cloud/localstack/awssdkv2/KinesisSchedulerTest.java

```diff
@@ -0,0 +1,93 @@
+package cloud.localstack.awssdkv2;
+
+import cloud.localstack.Localstack;
+import cloud.localstack.awssdkv2.consumer.DeliveryStatusRecordProcessorFactory;
+import cloud.localstack.awssdkv2.consumer.EventProcessor;
+import cloud.localstack.docker.annotation.LocalstackDockerProperties;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.model.CreateStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.CreateStreamResponse;
+import software.amazon.awssdk.services.kinesis.model.PutRecordRequest;
+import software.amazon.awssdk.services.kinesis.model.PutRecordResponse;
+import software.amazon.kinesis.common.ConfigsBuilder;
+import software.amazon.kinesis.coordinator.Scheduler;
+import software.amazon.kinesis.metrics.NullMetricsFactory;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+@LocalstackDockerProperties(ignoreDockerRunErrors = true)
+public class KinesisSchedulerTest extends PowerMockLocalStack {
+  String streamName = "test" + UUID.randomUUID().toString();
+  String workerId = UUID.randomUUID().toString();
+
+  @Before
+  public void mockServicesForScheduler() {
+    System.setProperty(SdkSystemSetting.CBOR_ENABLED.property(), "false");
+    PowerMockLocalStack.mockCloudWatchAsyncClient();
+    PowerMockLocalStack.mockDynamoDBAsync();
+    PowerMockLocalStack.mockKinesisAsync();
+  }
+
+  @Test
+  public void schedulerTest() throws Exception {
+
+    KinesisAsyncClient kinesisAsyncClient = KinesisAsyncClient.create();
+    DynamoDbAsyncClient dynamoAsyncClient = DynamoDbAsyncClient.create();
+    CloudWatchAsyncClient cloudWatchAsyncClient = CloudWatchAsyncClient.create();
+
+    createStream(kinesisAsyncClient);
+    TimeUnit.SECONDS.sleep(5);
+
+    EventProcessor eventProcessor = new EventProcessor();
+    DeliveryStatusRecordProcessorFactory processorFactory = new DeliveryStatusRecordProcessorFactory(eventProcessor);
+
+    ConfigsBuilder configsBuilder = new ConfigsBuilder(streamName, streamName, kinesisAsyncClient, dynamoAsyncClient,
+        cloudWatchAsyncClient, workerId, processorFactory);
+    Scheduler scheduler = createScheduler(configsBuilder);
+
+    new Thread(scheduler).start();
+    TimeUnit.SECONDS.sleep(50);
```

Comment:
Looks like a pretty long sleep time - does it take that long for the scheduler to become available and active?